meson.build

```diff
@@ -1,5 +1,172 @@
-project('tracy')
+project('tracy', ['cpp'])
 
-tracy_dep = declare_dependency(include_directories : include_directories('.'))
+if get_option('tracy_enable')
+  add_project_arguments('-DTRACY_ENABLE', language : 'cpp')
+endif
+
+if get_option('tracy_on_demand')
+  add_project_arguments('-DTRACY_ON_DEMAND', language : 'cpp')
+endif
+
+if get_option('tracy_callstack')
+  add_project_arguments('-DTRACY_CALLSTACK', language : 'cpp')
+endif
+
+if get_option('tracy_no_callstack')
+  add_project_arguments('-DTRACY_NO_CALLSTACK', language : 'cpp')
+endif
+
+if get_option('tracy_no_callstack_inlines')
+  add_project_arguments('-DTRACY_NO_CALLSTACK_INLINES', language : 'cpp')
+endif
+
+if get_option('tracy_only_localhost')
+  add_project_arguments('-DTRACY_ONLY_LOCALHOST', language : 'cpp')
+endif
+
+if get_option('tracy_no_broadcast')
+  add_project_arguments('-DTRACY_NO_BROADCAST', language : 'cpp')
+endif
+
+if get_option('tracy_only_ipv4')
+  add_project_arguments('-DTRACY_ONLY_IPV4', language : 'cpp')
+endif
+
+if get_option('tracy_no_code_transfer')
+  add_project_arguments('-DTRACY_NO_CODE_TRANSFER', language : 'cpp')
+endif
+
+if get_option('tracy_no_context_switch')
+  add_project_arguments('-DTRACY_NO_CONTEXT_SWITCH', language : 'cpp')
+endif
+
+if get_option('tracy_no_exit')
+  add_project_arguments('-DTRACY_NO_EXIT', language : 'cpp')
+endif
+
+if get_option('tracy_no_sampling')
+  add_project_arguments('-DTRACY_NO_SAMPLING', language : 'cpp')
+endif
+
+if get_option('tracy_no_verify')
+  add_project_arguments('-DTRACY_NO_VERIFY', language : 'cpp')
+endif
+
+if get_option('tracy_no_vsync_capture')
+  add_project_arguments('-DTRACY_NO_VSYNC_CAPTURE', language : 'cpp')
+endif
+
+if get_option('tracy_no_frame_image')
+  add_project_arguments('-DTRACY_NO_FRAME_IMAGE', language : 'cpp')
+endif
+
+if get_option('tracy_delayed_init')
+  add_project_arguments('-DTRACY_DELAYED_INIT', language : 'cpp')
+endif
+
+if get_option('tracy_manual_lifetime')
+  add_project_arguments('-DTRACY_MANUAL_LIFETIME', language : 'cpp')
+endif
+
+if get_option('tracy_fibers')
+  add_project_arguments('-DTRACY_FIBERS', language : 'cpp')
+endif
+
+tracy_shared_libs = get_option('tracy_shared_libs')
```

Comment:
Meson has a builtin per-subproject option for this. `get_option('default_library')` can be one of `shared` or `static` or `both`